lib/game.rb

```diff
@@ -1,66 +1,133 @@
 class Game 
   attr_reader :player_1,
-              :player_2
+              :player_2,
+              :board,
+              :column_choices
 
   def initialize
-    @player_1 = ""
-    @player_2 = ""
+    @board = Board.new
+    @is_bot = false
+    @column_choices = ["A", "B", "C", "D", "E", "F", "G"]
+    @player_1_turn = ""
+    @player_2_turn = ""
+    @player_1_selection = ""
+    @turn_count = 1
+  end
+
+  def user_input
+    gets.chomp
+  end
+
+  def set_player1
+    loop do
+      print "\nPlayer 1, please enter your name.\n"
+      @board.player_1 = user_input
+      @board.player_1 != "" ? break : (print "\nSorry. Please try again.")
+    end
+  end
+
+  #Reuse above for setting player2 
+
+  def is_bot?
+    @is_bot
   end
 
   def set_bot
-    @player_2 ="Computer"
+    @is_bot = true
+    @board.player_2 = "Computer"
+  end
+
+  def bot_selection
+    @turn_count += 1
+    @column_choices = @column_choices.shuffle
+    @column_choices[0]
   end
 
-  def main_menu_message
+  def welcome_message
     "\nWelcome to CONNECT FOUR\nEnter p to play. Enter q to quit.\n"
   end
-  #^there due to limitation on testing methods with user input.
 
-  def main_menu
-    print main_menu_message
-    if gets.chomp == "q"
-      abort ":("
-    elsif gets.chomp == "p"
+  def main_menu_user_input
+    choice = user_input
+    if choice == "p"
       set_player1
+      set_bot
+    elsif choice == "q"
+      abort ":("
     else
       print "\nSorry, please try again.\n"
     end
   end
 
-  def main_menu_loop
-    loop do
-      main_menu
-      if @player1 != ""
-        break
-      end
-    end
+  # a = ("a" if foo) || ("b" if bar) || "c"
+
+  # if foo 'a' elsif bar 'b' else 'c' syntax to help me refactor later -Mostafa
```

Comment:
Are you trying to condense lines 50-56?

Reply:
I've already condensed my if conditions. If you have tops on condensing them further, I'd appreciate the feedback. This is just as a reminder temporarily. Will be removing when we clean up the code, but since I used it a lot, I just kept it there as a reference as I'm working.

Reply:
gotcha ok. I wasn't sure what this was referring too